core/jvm/src/test/scala/scalaz/zio/stream/StreamSpec.scala

```diff
@@ -374,4 +403,42 @@ class StreamSpec(implicit ee: org.specs2.concurrent.ExecutionEnv)
         l   <- ref.get
       } yield l.reverse must_=== (0 to 10).toList
     )
+
+  private def bracket =
+    unsafeRun(
+      for {
+        done <- Ref.make(false)
+        iteratorStream = Stream.bracket(UIO(Iterator.range(0, 3)))(_ => done.set(true)) { it =>
+          if (it.hasNext) UIO(Some(it.next))
+          else UIO(None)
+        }
+        result <- iteratorStream.run(Sink.collect[Int])
+      } yield result must_=== List(0, 1, 2)
```

Comment:
There is no assertion on value of `done` after stream is run.

Reply:
Right! Thanks.